File: inject.js
const fs = require('fs');
const path = require('path');

const scriptTag = '<script src="wordbank-utils.js"></script>';
const dir = __dirname;

fs.readdir(dir, (err, files) => {
    if (err) throw err;

    const htmlFiles = files.filter(f => f.endsWith('.html') && f !== 'index.html' && f !== 'wordbank.html');

    htmlFiles.forEach(file => {
        const filePath = path.join(dir, file);
        let content = fs.readFileSync(filePath, 'utf8');

        if (!content.includes(scriptTag)) {
            if (content.includes('</body>')) {
                content = content.replace('</body>', `${scriptTag}\n</body>`);
            } else {
                content += `\n${scriptTag}`;
            }
            fs.writeFileSync(filePath, content, 'utf8');
            console.log(`Injected into ${file}`);
        } else {
            console.log(`Already injected in ${file}`);
        }
    });
    console.log('Done injecting!');
});

File: inject.py
import os
import glob

# Get all HTML files in the directory
html_files = glob.glob('*.html')

script_tag = '<script src="wordbank-utils.js"></script>'

for file_path in html_files:
    if file_path in ['index.html', 'wordbank.html']:
        continue
        
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
        
    if script_tag not in content:
        # Insert before </body> if it exists, otherwise append to end
        if '</body>' in content:
            new_content = content.replace('</body>', f'{script_tag}\n</body>')
        else:
            new_content = content + f'\n{script_tag}'
            
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Injected into {file_path}")
    else:
        print(f"Already injected in {file_path}")

print("Done injecting!")
